### ISSUE TITLE
Gold Tests Have Bit-Rotted

### ISSUE THREAD

**Issue by jordanthayer** [draperlaboratory/kittel-koat#6]
We've changed the output format of KoAT in a few different ways since last the golden tests were run.  The major difference comes with the way we rewrite all of the rules s.t. they all have the same arity.  Variables are renamed from x, y, z to ar_0, ar_1, ar_i, ... ar_n.  

It looks like the tests haven't been routinely run since we made the decision to always apply the arity fixing code, sometime in early February.

**Comment by jordanthayer** [comment]
CompareComplexity helps a lot here.  We aren't completely relying on diff, so we've got a little better behavior.  Currently it looks like the following files exhibit extremely different behavior (changes in order of magnitude in run time complexity) between the last time we ran the gold tests and the tip of the repository:

- branch-to-cycle
- exp1 through exp6
- exp13
- sep2
- test-parser
- test2 - 4

**Comment by jordanthayer** [comment]
Branch-to-cycle was a false positive.  If we have APRON turned on, we get the same result as the old tests.

test-parser and tests-2-4 are ok.  The new result is tighter.  There's a bug in either our checker or in the logic that runs the checker.

exp* are all equivalent.  I think the comparison function I wrote doesn't know how to deal with pow(), and gets the wrong results in this case.

sep2 is the only real difference.  Our version of koat gets no solution, whereas the version from which the gold standards were generated has an answer.

**Comment by jordanthayer** [comment]
Last commit merges in the fix_tests branch.  This adds a proper parser for complexity results, and a comparison of the parsed results.  Now we have a program telling us that the only different file is sep2, rather than me saying so.

**Comment by jordanthayer** [comment]
The difference in behavior is a result of our pruning the Cmeta branch out of KoAT.  Initially, Koat would try and determine if the program was unary, and run either cmeta's solver or cintmeta as appropriate.  We pruned away cmeta to get weighting in place faster.

The results we have are from a time before weighting was implemented, so that's why we have a result we can't replicate any more.  I'm going to mark the issue as closed now, but add another issue relating to rehabilitating the cmeta path in our fork of koat.
